server/connector.go

```diff
@@ -189,12 +193,45 @@ func (c *connectorImpl) findAndConnectBackend(ctx context.Context, frontendConn
 	c.metrics.ActiveConnections.Add(1)
 	defer c.metrics.ActiveConnections.Add(-1)
 
+	// PROXY protocol implementation
+	if c.sendProxyProto {
+		remoteHostStr, _, _ := net.SplitHostPort(backendHostPort)
+		sourceAddrStr, sourcePortStr, _ := net.SplitHostPort(clientAddr.String())
+		sourcePort, _ := strconv.Atoi(sourcePortStr)
+
+		header := &proxyproto.Header{
+			Version:           2,
+			Command:           proxyproto.PROXY,
+			TransportProtocol: proxyproto.TCPv4,
+			SourceAddr: &net.TCPAddr{
+				IP:   net.ParseIP(sourceAddrStr),
+				Port: sourcePort,
+			},
+			DestinationAddr: &net.TCPAddr{
+				IP: net.ParseIP(remoteHostStr),
```

Comment:
@Gabitzuu a user is reporting a problem with this feature

```
level=info msg="Got connection" client="xxx.xxx.xxx.xxx:48064"
level=info msg="Connecting to backend" backendHostPort="velocity:25577" client="xxx.xxx.xxx.xxx:48064" server=xxx
level=error msg="Failed to write PROXY header" clientAddr="xxx.xxx.xxx.xxx:48064" destAddr=":0" error="proxyproto: invalid address"
```

and I'm now wondering why a hostname, "velocity" in this case, is being parsed as an IP address. Did this feature assume IP addresses were manually configured as the backend routes?

Reply:
Hi and sorry for the late response. It's been a long time since I've wrote this implementation. I think that's what I've intended, to only use IP addresses but not hostnames.
I was a Golang noobie back then so this feature is not written in the best way.

Reply:
Thanks for getting to me. That's totally fine and just wanted to check I wasn't missing something. I'll change this to use either hostname or IP address.